django_rdkit/models/fields.py

```diff
@@ -58,11 +67,11 @@ def get_prep_value(self, value):
         # convert the Molecule instance to the value used by the
         # db driver
         if isinstance(value, six.string_types):
-            # The string case. A SMILES is assumed.
-            value = Chem.MolFromSmiles(str(value))
+            # The string case. Don't assume SMILES
+            value = self.to_python(str(value))
         if isinstance(value, Chem.Mol):
             value = six.memoryview(value.ToBinary())
-        return value
+        return Func(value, function='mol_from_pkl')
```

Comment:
there are a couple more things I'd like to check here (e.g. I'm not sure if sql expression should be also excluded).

Reply:
My understanding is that since this function is a member of the MolField object, 'value' should contain either a mol object or a str which was entered by the user, but which has not yet been converted to a mol object via clean()->to_python(). If this is correct, then this function should handle both cases. I'm still learning django, so feel free to teach me a few things.

Reply:
I understand - makes sense. The extra tests have been merged into the pr.